### PR TITLE
contact hyperlink issue fixed

### PR DESCRIPTION
**Pull request by devsekar-1992**
I fixed the issue for  this open issue https://github.com/nextcloud/server/issues/24807

### CODE REVIEW

**Review by szaimen — APPROVED** [review]
Works! Thanks for fixing this :)

**Review by Pytal — COMMENTED** [review]
Could we handle the case when `contact.topAction` is not defined?

**Comment by szaimen** [comment]
> Could we handle the case when `contact.topAction` is not defined?

@devsekar-1992?

**Comment by Pytal** [comment]
@devsekar-1992 from line 30 of https://github.com/devsekar-1992/server/blob/contact-popover-issue/core/src/OC/contactsmenu/contact.handlebars something like the below should work
```handlebars
{{else if contact.topAction}}
	<a class="body" href="{{contact.topAction.hyperlink}}">
		<div class="full-name">{{contact.fullName}}</div>
		<div class="last-message">{{contact.lastMessage}}</div>
		<div class="email-address">{{contact.emailAddresses}}</div>
	</a>
	<a class="top-action" href="{{contact.topAction.hyperlink}}" title="{{contact.topAction.title}}">
		<img src="{{contact.topAction.icon}}" alt="{{contact.topAction.title}}">
	</a>
{{else}}
	<div class="body">
		<div class="full-name">{{contact.fullName}}</div>
		<div class="last-message">{{contact.lastMessage}}</div>
		<div class="email-address">{{contact.emailAddresses}}</div>
	</div>
{{/if}}
{{#if contact.hasTwoActions}}
```

**Comment by devsekar-1992** [comment]
> @devsekar-1992 from line 30 of https://github.com/devsekar-1992/server/blob/contact-popover-issue/core/src/OC/contactsmenu/contact.handlebars something like the below should work
> 
> ```
> {{else if contact.topAction}}
> 	<a class="body" href="{{contact.topAction.hyperlink}}">
> 		<div class="full-name">{{contact.fullName}}</div>
> 		<div class="last-message">{{contact.lastMessage}}</div>
> 		<div class="email-address">{{contact.emailAddresses}}</div>
> 	</a>
> 	<a class="top-action" href="{{contact.topAction.hyperlink}}" title="{{contact.topAction.title}}">
> 		{{contact.topAction.title}}
> 	</a>
> {{else}}
> 	<div class="body">
> 		<div class="full-name">{{contact.fullName}}</div>
> 		<div class="last-message">{{contact.lastMessage}}</div>
> 		<div class="email-address">{{contact.emailAddresses}}</div>
> 	</div>
> {{/if}}
> {{#if contact.hasTwoActions}}
> ```

Ok I will do

File: core/src/OC/contactsmenu/contact.handlebars [hunk 1]
```diff
@@ -27,18 +27,22 @@
...
 	{{/if}}
+{{else if contact.topAction}}
+	<a class="body" href="{{contact.topAction.hyperlink}}">
+		<div class="full-name">{{contact.fullName}}</div>
+		<div class="last-message">{{contact.lastMessage}}</div>
+		<div class="email-address">{{contact.emailAddresses}}</div>
+	</a>
+  <a class="top-action" href="{{contact.topAction.hyperlink}}" title="{{contact.topAction.title}}">
+  <img src="{{contact.topAction.icon}}" alt="{{contact.topAction.title}}">
+</a>
```
Comment (Pytal) [inline core/src/OC/contactsmenu/contact.handlebars:38]:
```suggestion
	<a class="body" href="{{contact.topAction.hyperlink}}">
		<div class="full-name">{{contact.fullName}}</div>
		<div class="last-message">{{contact.lastMessage}}</div>
		<div class="email-address">{{contact.emailAddresses}}</div>
	</a>
  	<a class="top-action" href="{{contact.topAction.hyperlink}}" title="{{contact.topAction.title}}">
  		<img src="{{contact.topAction.icon}}" alt="{{contact.topAction.title}}">
	</a>
```
Indentation fix

**Review by Pytal — CHANGES_REQUESTED** [review]

**Review by devsekar-1992 — COMMENTED** [review]
Thanks for indent

**Review by Pytal — APPROVED** [review]

**Comment by szaimen** [comment]
@devsekar-1992 could you please fix dco, rebase your branch on top of master and compile and commit the js? Thank you! :)

**Comment by devsekar-1992** [comment]
> @devsekar-1992 could you please fix dco, rebase your branch on top of master and compile and commit the js? Thank you! :)

I did dco but still showing error

**Comment by szaimen** [comment]
> I did dco but still showing error

image

---

### How to fix it:


To add your Signed-off-by line to every commit in this branch:

Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
In your local branch, run: `git rebase HEAD~9 --signoff`
Force push your changes to overwrite the branch: `git push --force-with-lease origin contact-popover-issue`

**Comment by devsekar-1992** [comment]
> > I did dco but still showing error
> 
> image
> 
> ### How to fix it:
> To add your Signed-off-by line to every commit in this branch:
> 
> Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally). In your local branch, run: `git rebase HEAD~9 --signoff` Force push your changes to overwrite the branch: `git push --force-with-lease origin contact-popover-issue`
After rebase and push ,shows everything  up-to-date

**Comment by devsekar-1992** [comment]
@szaimen  please use below pull request

https://github.com/nextcloud/server/pull/31824/

**Comment by szaimen** [comment]
Superseded by https://github.com/nextcloud/server/pull/31824

**Pull request closed without merging**